### PR TITLE
[SimpleConsumer] Fix legacy SimpleConsumer with compressed messages

### PR DESCRIPTION
**Pull request by paulcavallaro**
Summary:

Fix legacy SimpleConsumer with compressed messages. Decompress messages if they are compressed and add a test for the new logic.

Test Plan:

KAFKA_VERSION=0.9.0.1 tox -e py27 test/test_consumer_integration.py

### CODE REVIEW

File: kafka/consumer/simple.py [hunk 1]
```diff
@@ -433,12 +434,26 @@ def _fetch(self):
...
-                        continue
-                    # Put the message in our queue
-                    self.queue.put((partition, message))
-                    self.fetch_offsets[partition] = message.offset + 1
+                for off_and_msg in resp.messages:
+                    (offset, message) = off_and_msg
+                    # Put the messages in our queue
+                    if message.is_compressed():
+                        inner_messages = message.decompress()
+                        for (offset, _msg_size, inner_msg) in inner_messages:
```
Comment (dpkp) [inline kafka/consumer/simple.py]: might call this inner_offset to avoid the redefinition

**Comment by dpkp** [comment]
You probably will need to handle this in KafkaProtocol.decode_fetch_response

**Comment by paulcavallaro** [comment]
Looks like one of the tests failed due to spurious `Exception: Failed to start KafkaInstance before max_timeout` is there a way to retry that test?

**Comment by dpkp** [comment]
thanks!

**Pull request merged**